Tema2.js

```diff
@@ -0,0 +1,102 @@
+//problem 1
+console.log(problem1(3));
+function problem1(number) {
+    if (number % 2 != 0)
+        return number * number;
+    else
+        return number - 2;
+}
+
+
+//problem 2
+const savePrimes = primes => {
+    //console.log(savePrimes[1]);
+    const arrayPrimes = new Array;
+    let dimension = primes.length;
+    let contor = 0;
+
+    for (let i = 0; i < dimension; i++) {
+        let prim = true;
+        if (primes[i] < 2)
+            continue;
+        for (let div = 2; div < primes[i]; div++) {
+            if (primes[i] % div == 0)
+                prim = false;
+
+        }
+        if (prim) {
+            arrayPrimes[contor++] = primes[i];
+        }
+    }
+    console.log(arrayPrimes);
+}
+
+const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+savePrimes(arr);
+
+
+//problem 3 (IIFE)
+const saveStrings = [1, '1', 'salut', 2, 6, true, 'true'];
+((arrStrings) => {
+    const justStrings = new Array;
+    let dim = arrStrings.length;
+    let k = 0;
+
+        arrStrings.forEach(item => {
+            if(typeof(item) === 'string') 
+              justStrings[k++] = item;
+        });
+        console.log(justStrings);
+        
+})(saveStrings);
+
+
+//problem 4 (arrow function)
+const sum = SumOfDigits => {
+    Number(SumOfDigits); //convert strings to number
```

Comment:
Problema 4 are o logica corecta, vreau doar sa mentionez ca _Number(SumOfDigits);_ returneaza _SumOfDigits_ ca si numar, dar nu il muteaza, deci _SumOfDigits_  ramane ca si string pe parcursul functiei (poti verifica facand un console.log(typeof _SumOfDigits_).

O alta abordare ar fi sa parcurgi numarul scris sub forma de string ca si un array obisnuit:
"1978"[0] = 1
"1978"[1] = 9
"1978"[2] = 7
"1978"[3] = 8
Astfel il poti itera cu un for fara sa mai fie nevoie sa imparti la 10

Reply:
Aici daca ai timp, incearca sa abordezi problema folosind cele mentionate in comment ^.